app/__init__.py

```diff
@@ -20,19 +20,89 @@ def create_app(config_name):
 	app.config['SQLALCHEMY_TRACK_MODIFICATIONS'] = False
 	db.init_app(app)
 
+	@app.route('/campsites/<int:id>', methods=['GET', 'PUT', 'DELETE'])
+	def campsite_manipulation(id, **kwargs):
+		campsite = Campsite.query.filter_by(id=id).first()
+		if not campsite:
+			abort(404)
+
+		if request.method == 'DELETE':
+			campsite.delete()
+			return {
+				"message": "campsite {} deleted successfully".format(campsite.id)
+			}, 200
+
+		elif request.method == 'PUT':
+			name = str(request.data.get('name', campsite.name))
```

Comment:
Finding a way to refactor these methods should be one of our first refactor points. Maybe we can make a model method for them.